libraries/AP_Baro/AP_Baro_DPS280.cpp

```diff
@@ -0,0 +1,232 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  DPS280 barometer driver
+ */
+
+#include "AP_Baro_DPS280.h"
+
+#include <utility>
+#include <stdio.h>
+
+extern const AP_HAL::HAL &hal;
+
+#define DPS280_REG_PRESS  0x00
+#define DPS280_REG_TEMP   0x03
+#define DPS280_REG_PCONF  0x06
+#define DPS280_REG_TCONF  0x07
+#define DPS280_REG_MCONF  0x08
+#define DPS280_REG_CREG   0x09
+#define DPS280_REG_ISTS   0x0A
+#define DPS280_REG_FSTS   0x0B
+#define DPS280_REG_RESET  0x0C
+#define DPS280_REG_PID    0x0D
+#define DPS280_REG_COEF   0x10
+#define DPS280_REG_CSRC   0x28
+
+#define DPS280_WHOAMI 0x10
+
+
+AP_Baro_DPS280::AP_Baro_DPS280(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> _dev)
+    : AP_Baro_Backend(baro)
+    , dev(std::move(_dev))
+{
+}
+
+AP_Baro_Backend *AP_Baro_DPS280::probe(AP_Baro &baro,
+                                       AP_HAL::OwnPtr<AP_HAL::Device> _dev)
+{
+    if (!_dev) {
+        return nullptr;
+    }
+
+    AP_Baro_DPS280 *sensor = new AP_Baro_DPS280(baro, std::move(_dev));
+    if (!sensor || !sensor->init()) {
+        delete sensor;
+        return nullptr;
+    }
+    return sensor;
+}
+
+/*
+  handle bit width for 16 bit config registers
+ */
+void AP_Baro_DPS280::fix_config_bits16(int16_t &v, uint8_t bits) const
```

Comment:
why return per reference? if so, why not making it static? This function is independent on class members and should be static.